src/main/scala/chisel3/util/experimental/decode/decoder.scala

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.experimental.decode
+
+import chisel3._
+import chisel3.experimental.{ChiselAnnotation, annotate}
+import chisel3.util.{BitPat, pla}
+import chisel3.util.experimental.getAnnotations
+import firrtl.annotations.Annotation
+import logger.LazyLogging
+
+object decoder extends LazyLogging {
+  def apply(minimizer: Minimizer, input: UInt, truthTable: TruthTable): UInt = {
+    val minimizedTable = getAnnotations().collect {
+      case DecodeTableAnnotation(_, in, out) => TruthTable(in) -> TruthTable(out)
+    }.toMap.getOrElse(
+      {
+        logger.trace(s"""Decoder Cache Hit!
+                        |${truthTable.table}
+                        |""".stripMargin)
+        truthTable
+      }, {
+        val startTime = System.nanoTime()
+        val minimizedTable = minimizer.minimize(truthTable)
+        val totalTime = System.nanoTime() - startTime
+        val totalTimeInSeconds = totalTime / 1e9
+        val info = f"Logic Minimize with $minimizer finished in ${totalTimeInSeconds} second"
+        if (totalTimeInSeconds > 5)
```

Comment:
I don't like the idea of having a built-in timeout, especially with a hard-coded limit.  I suggest removing this entirely, but if you insist on keeping it, make the limit a parameter that defaults to infinity.

Reply:
Got it!